backend/app/websocket/persistence.py

```
@@ -0,0 +1,167 @@
+        
+        Args:
+            message: The message to persist
+            
+        Returns:
+            bool: True if message was successfully persisted, False otherwise
+        """
+        try:
+            with self._lock:
+                # Check if we've reached capacity
+                if len(self.storage) >= self.max_messages:
+                    if message.id not in self.storage:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)